src/hotspot/share/opto/escape.cpp

```diff
@@ -2819,14 +2819,13 @@ int ConnectionGraph::find_init_values_null(JavaObjectNode* pta, PhaseValues* pha
         offsets_worklist.append(offset);
         Node* value = nullptr;
         if (ini != nullptr) {
-          // StoreP::memory_type() == T_ADDRESS
```

Comment:
The value of this comment is debatable, but I think it is best to not remove it in this RFE (just rename `memory_type` to `value_basic_type`), to preserve the original scope of the RFE.